api/v1beta1/types.go

```diff
@@ -509,6 +509,9 @@ const (
 
 	// ServicePrincipalCertificate represents a service principal using a certificate as secret.
 	ServicePrincipalCertificate IdentityType = "ServicePrincipalCertificate"
+
+	// WorkloadIdentity represents a WorkloadIdentity.
+	WorkloadIdentity IdentityType = "WorkloadIdentity"
```

Comment:
wonder if this is an appropriate name for this identity "type" or if the name should be based more on the Azure identity resource (`FederatedIdentityCredential`) as opposed to the k8s project that enables this (`WorkloadIdentity`) as it's more of an implementation detail.

Also do we foresee Workload Identity supporting more than one "type" in the future? For example, it looks like you could already use either AD application or User Assigned Identity https://azure.github.io/azure-workload-identity/docs/topics/federated-identity-credential.html#federated-identity-credential-for-a-user-assigned-managed-identity, so we should be able to differentiate between those (unless the code to use them in CAPZ is the same)? @aramase thoughts?

Reply:
> Also do we foresee Workload Identity supporting more than one "type" in the future? For example, it looks like you could already use either AD application or User Assigned Identity https://azure.github.io/azure-workload-identity/docs/topics/federated-identity-credential.html#federated-identity-credential-for-a-user-assigned-managed-identity, so we should be able to differentiate between those (unless the code to use them in CAPZ is the same)? @aramase thoughts?

The code to use workload identity federation would be the same irrespective of the underlying Azure AD object (application or managed identity) it's tied to. In terms of the name, I think [Workload Identity Federation](https://learn.microsoft.com/en-us/azure/active-directory/develop/workload-identity-federation) is used commonly.

Reply:
Sure. Like @aramase said -- it will really not matter if we use an AD application or User Assigned Identity as far as CAPZ implementation is concerned. 
Also I consider `Workload Identity Federation` as one of the entities/workflows of the entire `Workload Identity` thing. 
And `Workload Identity` will be there for other providers too ( if not yet there ) though the implementations and terminology under the hood may differ. 

So `WorkloadIdentity` looks simple, and intuitive to me from a user's POV. 

Just to take decision -- we are deciding b/w `WorkloadIdentity` vs `WorkloadIdentityFederation` , right?